src/main/scala/tsiHost/TsiHostPeriphery.scala

```diff
@@ -0,0 +1,67 @@
+package testchipip
+
+import chisel3._
+import chisel3.util._
+
+import freechips.rocketchip.config.{Parameters, Field}
+import freechips.rocketchip.subsystem._
+import freechips.rocketchip.diplomacy._
+import freechips.rocketchip.tilelink._
+import freechips.rocketchip.util._
+import freechips.rocketchip.amba.axi4._
+
+/**
+ * Configuration parameter to configure a set of TSI Host Widgets
+ */
+case object PeripheryTSIHostKey extends Field[Seq[TSIHostParams]](Nil)
+
+/**
+ * Trait to create a set of TSI Host Widgets
+ */
+trait HasPeripheryTSIHostWidget { this: BaseSubsystem =>
+  val tsiHostWidgetNodes = p(PeripheryTSIHostKey).map { ps =>
+    // TODO: Measure BW if using sbus instead of pbus
```

Comment:
I think this was just a random note to determine the best place to attach the MMIO node. I think it being on the PBUS is probably good enough since we probably aren't looking to do a lot of sustained TSI reqs.